lib/crypto.ml

```diff
@@ -73,6 +76,23 @@ let sequence_buf seq =
   BE.set_uint64 buf 0 seq ;
   buf
 
+let aead_nonce nonce seq =
+  let s =
+    let l = Cstruct.len nonce in
+    let s = sequence_buf seq in
+    let pad = Cs.create (l - 8) in
+    pad <+> s
+  in
+  Cs.xor nonce s
+
+let adata_1_3 len =
+  let buf = Cstruct.create 5 in
+  Cstruct.set_uint8 buf 0 (Packet.content_type_to_int Packet.APPLICATION_DATA) ;
```

Comment:
I had to dig around a bit to understand what the two `\x03\x03` here are (the TLS_1_3 version tuple); it looks to me like this function is implementing https://www.rfc-editor.org/rfc/rfc8446.html#section-5.2 (page 81), which would be good to have documented?

This function seems very similar to `pseudo_header` (which implements https://tools.ietf.org/html/rfc5246#section-6.2.3.3 (top of page 25)) - that would also be good to have documented!